test/cluster/test_sstable_set.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (C) 2025-present ScyllaDB
+#
+# SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+#
+
+import asyncio
+import pytest
+import time
+import logging
+from test.pylib.manager_client import ManagerClient
+from test.pylib.util import wait_for_cql_and_get_hosts
+from test.cluster.util import create_new_test_keyspace
+from test.cluster.conftest import skip_mode
+
+logger = logging.getLogger(__name__)
+
+@pytest.mark.parametrize("mode", ['vnode', 'tablet'])
+@skip_mode('release', 'error injections are not supported in release mode')
+@pytest.mark.asyncio
+async def test_partitioned_sstable_set(manager: ManagerClient, mode):
+    cfg = {
+        'tablets_mode_for_new_keyspaces': 'enabled',
+    }
+
+    cmdline = ['--smp=1']
+    server = await manager.server_add(config=cfg, cmdline=cmdline)
+    await manager.api.disable_tablet_balancing(server.ip_addr)
+
+    cql = manager.get_cql()
+    await wait_for_cql_and_get_hosts(cql, [server], time.time() + 60)
+
+    if mode == 'tablet':
+        ks = await create_new_test_keyspace(cql, "WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1} AND tablets = {'initial': 4};")
+    else:
+        ks = await create_new_test_keyspace(cql, "WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1} AND tablets = {'enabled': 'false'};")
```

Comment:
let's drop te keyspace at the end of the test to facilitate reuse of the cluster in the future.
Or you could use `async with new_test_keyspace` and just format the `WITH replication` extra stuff like
`f"WITH replication = {{'class': 'NetworkTopologyStrategy', 'replication_factor': 1}} AND tablets = {{{'\'initial\': 4' if mode=='tablet' else '\'enabled\': \'false\''}}}"`